HdfsAccessor.go

```diff
@@ -118,14 +118,17 @@ func (this *hdfsAccessorImpl) connectToNameNodeImpl(nnAddr string) (*hdfs.Client
 
 // Opens HDFS file for reading
 func (this *hdfsAccessorImpl) OpenRead(path string) (ReadSeekCloser, error) {
-	client, err1 := this.ConnectToNameNode()
-	if err1 != nil {
-		return nil, err1
+	// Blocking read. This is to reduce the connections pressue on hadoop-name-node
+	this.MetadataClientMutex.Lock()
+	defer this.MetadataClientMutex.Unlock()
+	if this.MetadataClient == nil {
```

Comment:
Can we wrap this check and call into its own method so we just call `this.OpenMetadataClient(path)` and it magically connects up a new client if we don't have one already? This code feels like it's leaking abstraction fluid.